### PR TITLE
v0.7.0 with support for cuda12

### LINKED ISSUE #29: Fix Linux CUDA 11.8 GCC 11 builds
Recently when adding CUDA 11.8 builds in PR ( https://github.com/conda-forge/cudadecon-feedstock/pull/26 ), we ran into some issues with the Linux builds. On further investigation, it appears that the issue was related to the use of GCC 11 in the Linux CUDA 11.8 builds

Looking at the result from [CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=813158&view=logs&jobId=4f922444-fdfe-5dcf-b824-02f86439ef14&j=4f922444-fdfe-5dcf-b824-02f86439ef14 ) (also [attached log]( https://github.com/conda-forge/cudadecon-feedstock/files/13386091/cudadecon-feedstock_20231028.2_linux_64_c_compiler_version11cuda_compilernvcccuda_compiler_version11.8cxx_compiler_version11.txt ) for posterity), we saw errors like this one:

```cxx
/home/conda/feedstock_root/build_artifacts/cudadecon_1698498913015/_build_env/x86_64-conda-linux-gnu/include/c++/11.4.0/type_traits:79:52: error: redefinition of 'constexpr const _Tp std::integral_constant<_Tp, __v>::value'
   79 |   template<typename _Tp, _Tp __v>
      |                                                    ^                           
/home/conda/feedstock_root/build_artifacts/cudadecon_1698498913015/_build_env/x86_64-conda-linux-gnu/include/c++/11.4.0/type_traits:67:29: note: 'constexpr const _Tp value' previously declared here
   67 |       static constexpr _Tp                  value = __v;
      |      
```

This error did not occur with GCC 10. So something changed between GCC 10 & 11. Maybe the compiler got stricter and so is catching an error? Would need further investigation to know for sure

For now we have pinned to GCC 10 in the CUDA 11.8 builds as a workaround, which allows those to pass. Have highlighted the relevant lines below

https://github.com/conda-forge/cudadecon-feedstock/blob/3c13e13c595117ab0ca5bf81467312f7bf8d6e82/recipe/meta.yaml#L19-L24

Once the underlying issue is resolved we should remove the workaround by reverting commit: https://github.com/conda-forge/cudadecon-feedstock/commit/4cb126315511cfc1528d13e498c53ad5af3c54b8

### PR DESCRIPTION
**Pull request by tlambert03**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes https://github.com/conda-forge/cudadecon-feedstock/issues/29
Closes https://github.com/conda-forge/cudadecon-feedstock/pull/27

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,36 +1,32 @@
...
     - {{ stdlib('c') }}
-    # For Linux CUDA 11.8 builds, pin to GCC 10 to workaround issues with GCC 11.
-    # xref: https://github.com/conda-forge/cudadecon-feedstock/issues/29
-    - {{ compiler('c') }}                          # [not (linux and cuda_compiler_version == "11.8")]
-    - {{ stdlib("c") }}                            # [not (linux and cuda_compiler_version == "11.8")]
-    - {{ compiler('cxx') }}                        # [not (linux and cuda_compiler_version == "11.8")]
-    - {{ c_compiler }}_{{ target_platform }} 10    # [linux and cuda_compiler_version == "11.8"]
-    - {{ cxx_compiler }}_{{ target_platform }} 10  # [linux and cuda_compiler_version == "11.8"]
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
```
Comment (jakirkham) [inline recipe/meta.yaml:20]: Am so happy to see this going away 😂

**Review by jakirkham — COMMENTED** [review]
Thanks Talley! 🙏

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

(just in case it needs one more)

**Comment by jakirkham** [comment]
FWIW LGTM

If you want to have the bot merge after CI passes, can use the `automerge` label

**Comment by jakirkham** [comment]
JFYI Azure is having and outage, which is why this is stuck atm

Would just wait for it to resolve. CI should hopefully pick up then and finish. If not, we can always restart

More details in issue: https://github.com/conda-forge/status/issues/184

**Comment by tlambert03** [comment]
Oh good to know, I was wondering wtf.

Yeah I'll check in later and bump it

**Comment by jakirkham** [comment]
Yeah we just got really unlucky 😬

Thanks Talley. Sorry for the hangup 😞

**Comment by jakirkham** [comment]
Looks like Azure built on `main`

Please recheck though

**Pull request merged**